### ISSUE TITLE
(dev) empty line at the end of file

### ISSUE THREAD

**Issue by topchyan** [sciurius/chordpro#30]
Following example without a blank line at the end of file...
```
{start_of_verse: Last Verse}
This is the song that has doesn't end,
It just goes on and on, my friend,
Some people started singing it not knowing what it was,
And they'll continue singing it forever just because...
{end_of_verse}
```
Produces this warning: 
        **line 408, Unterminated context in song: verse
	And they'll continue singing it forever just because...**

But with an empty line at the end of file it processes it normally

**Comment by sciurius** [comment]
I cannot reproduce this with 0.91 (release) and 0.910_014 (dev).

Can you provide a test file?

**Comment by topchyan** [comment]
The first statement file error.pro shows the message, but the second one no_error.pro does not. The difference between the two files is one last line.

```
> chordpro "/home/avetik/error.pro" -v         -cfg ~/Projects/ChordPro/custom.json         -cfg ~/Projects/ChordPro/fonts-liberation.json         -cfg ~/Projects/ChordPro/ukulele-ly-plus.json         --output="/home/avetik/error.pro.pdf" && xreader "/home/avetik/error.pro.pdf"
Config: /home/avetik/Projects/ChordPro/custom.json
Config: /home/avetik/Projects/ChordPro/fonts-liberation.json
Config: /home/avetik/Projects/ChordPro/ukulele-ly-plus.json
"/home/avetik/error.pro", line 48, Unterminated context in song: verse
	Но в этот час стало всем теплей.
>Exit code: 0
```

```
>chordpro "/home/avetik/no_error.pro" -v         -cfg ~/Projects/ChordPro/custom.json         -cfg ~/Projects/ChordPro/fonts-liberation.json         -cfg ~/Projects/ChordPro/ukulele-ly-plus.json         --output="/home/avetik/no_error.pro.pdf" && xreader "/home/avetik/no_error.pro.pdf"
Config: /home/avetik/Projects/ChordPro/custom.json
Config: /home/avetik/Projects/ChordPro/fonts-liberation.json
Config: /home/avetik/Projects/ChordPro/ukulele-ly-plus.json
>Exit code: 0
```
[error.pro.txt](https://github.com/sciurius/chordpro/files/1510554/error.pro.txt)
[no_error.pro.txt](https://github.com/sciurius/chordpro/files/1510555/no_error.pro.txt)



[custom.json.txt](https://github.com/sciurius/chordpro/files/1510562/custom.json.txt)
[fonts-liberation.json.txt](https://github.com/sciurius/chordpro/files/1510563/fonts-liberation.json.txt)
[ukulele-ly-plus.json.txt](https://github.com/sciurius/chordpro/files/1510564/ukulele-ly-plus.json.txt)

**Comment by topchyan** [comment]
Sorry, closed the issue by mistake.

**Comment by sciurius** [comment]
There is a very old controversy whether a newline character is a line separator or a line terminator. Usually, newline is considered a line terminator, and text files are expected to contain a final newline. Many text oriented programs malfunction if the file does not contain that final newline.

The sample file "error.pro.txt" does not contain a final newline, while "no_error.pro.txt" does. ChordPro expects the lines to be newline terminated, so it does not see the last line ({end_of_verse}) and throws an error.

Despite my strong bias towards newline termination, I made a fix for this case.

**Comment by topchyan** [comment]
I use SciTE on Linux to edit source files, so all I saw was that this was caused by not having an empty line at the end of the text after a {end_of_verse} directive. Even though I figure it out, there might be some users for whom this is not very clear, so I believe eventually this issue would have come up anyway, so I think this fix is a plus, so thank you! I also think platform differences (Windows vs Mac vs Linux) can have an effect on line and file termination.
